### ISSUE TITLE
nakama.http_request doesn't work for RPC hook

### ISSUE THREAD

**Issue by jompu** [heroiclabs/nakama#211]
## Description
`nakama.http_request` doesn't work for RPC hook. I would like to use Lua runtime to write tests for my own Lua code. Particularly to test RPC hooks and verify results from database etc.

## Steps to Reproduce

Create a module with following code:
```Lua
local nk = require("nakama")

-- Called from other server.
local function serverToServerEndpoint(context, payload)
    local message = nk.json_decode(payload)
    -- Does something
    return nk.json_encode({["context"] = context})
end

nk.register_rpc(serverToServerEndpoint, "serverToServerEndpoint")

local function runTests()
    local url = "http://127.0.0.1:7350/v2/rpc/serverToServerEndpoint?http_key=defaultkey"
    local method = "POST"
    local headers = {
        ["Content-Type"] = "application/json",
        ["Accept"] = "application/json"
    }
    local content = nk.json_encode({})

    local success, code, headers, body = pcall(
        nk.http_request, 
        url, 
        method, 
        headers, 
        content
    )

    if (not success) then
        nk.logger_error(("Failed %q"):format(code))
        error(code)
    elseif (code ~= 200) then
        nk.logger_error(("Failed %q %q"):format(code, body))
    else
        nk.logger_info(("Success %q %q"):format(code, body))

        -- Verify from database that serverToServerEndpoint worked correctly.
    end
end

local function testEndpoint(context, payload)
    local message = nk.json_decode(payload)
    
    runTests()

    return nk.json_encode({["context"] = context})
end

nk.register_rpc(testEndpoint, "testEndpoint")
```

Then call testEndpoint with a client outside Nakama.

## Expected Result
Print success

## Actual Result
```
"Failed \"400\" \"{\\\"error\\\":\\\"json: cannot unmarshal object into Go value of type string\\\",\\\"code\\\":3}\"","stacktrace":"github.com/heroiclabs/nakama/server.(*NakamaModule).loggerError\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/runtime_nakama_module.go:1225\ngithub.com/heroiclabs/nakama/server.(*NakamaModule).(github.com/heroiclabs/nakama/server.loggerError)-fm\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/runtime_nakama_module.go:144\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.callGFunction\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/vm.go:123\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.init.2.func26\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/vm.go:704\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.mainLoopWithContext\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/vm.go:60\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.(*LState).callR\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/state.go:876\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.(*LState).Call\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/state.go:1610\ngithub.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua.(*LState).PCall\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/github.com/yuin/gopher-lua/state.go:1673\ngithub.com/heroiclabs/nakama/server.(*Runtime).invokeFunction\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/runtime.go:249\ngithub.com/heroiclabs/nakama/server.(*Runtime).InvokeFunction\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/runtime.go:225\ngithub.com/heroiclabs/nakama/server.(*ApiServer).RpcFunc\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/api_rpc.go:61\ngithub.com/heroiclabs/nakama/api._Nakama_RpcFunc_Handler.func1\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/api/api.pb.go:4503\ngithub.com/heroiclabs/nakama/server.apiInterceptorFunc.func1\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/server/api.go:199\ngithub.com/heroiclabs/nakama/api._Nakama_RpcFunc_Handler\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/api/api.pb.go:4505\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).processUnaryRPC\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:923\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).handleStream\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:1148\ngithub.com/heroiclabs/nakama/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1\n\t/Users/andrei/Development/go/src/github.com/heroiclabs/nakama/vendor/google.golang.org/grpc/server.go:637"}

```

## Context
Called with python:
```Python
import requests

payload = """ { }
"""

headers = {
    "Accept": "application/json"
}

r = requests.post(
    "http://127.0.0.1:7350/v2/rpc/testEndpoint?http_key=defaultkey",
    json=payload,
    headers=headers
)
```

## Your Environment
- Nakama: 2.0.0+e77afd7
- Cockroach: 2.0.1
- Operating System and version: Fedora 28

**Comment by zaniar** [comment]
You can use:
```python
import requests
import json

payload = {}

requests.post('http://127.0.0.1:7350/v2/rpc/testEndpoint?http_key=defaultkey', data='"'+json.dumps(payload)+'"')
```

**Comment by zyro** [comment]
Aside from the working solution already posted, this is further resolved in 20d6ab3cfc2478c39498a4d261d23c9fd1906ef2 and will be part of the next release.

Hooks will accept an `unwrap` query parameter and receive/return any raw string payload without wrappers. The documentation will be updated to reflect this in time for the next release.
